core/src/main/java/io/grpc/internal/Http2ClientStream.java

```diff
@@ -205,9 +214,9 @@ private Status checkContentType(Metadata headers) {
     }
     contentTypeChecked = true;
     String contentType = headers.get(GrpcUtil.CONTENT_TYPE_KEY);
-    if (TEMP_CHECK_CONTENT_TYPE && !GrpcUtil.CONTENT_TYPE_GRPC.equalsIgnoreCase(contentType)) {
-      // Malformed content-type so report an error
-      return Status.INTERNAL.withDescription("invalid content-type " + contentType);
+    if (contentType != null && !GrpcUtil.isGrpcContentType(contentType)) {
```

Comment:
Ok, so we require that the first received HEADERS frame contains the content-type? Makes sense.